modules/library/coverage/src/main/java/org/geotools/coverage/grid/io/footprint/MultiLevelROIGeometryProvider.java

```diff
@@ -31,22 +31,51 @@
 
     private final double inset;
 
+    /** The FootprintInsetPolicy if any for inset management */
     private final FootprintInsetPolicy insetPolicy;
 
+    /** 
+     * The optional granuleBounds to be used as a default value when there is 
+     * no need to do ROI lookup against specific features.
+     * Specify that when dealing with single granules store  
+     */
+    private final Geometry imposedGranuleBounds;
```

Comment:
Indeed, in the beginning, I called that defaultGranuleBounds as the comment suggests. I have switched to "imposed" to indicate it is explicitated in the constructor. Maybe is misleading as you say... going back to a "default" "defaultGranuleBounds"